integration-tests/nitrite/src/test/java/org/apache/camel/quarkus/component/nitrite/it/NitriteTest.java

```diff
@@ -64,12 +98,15 @@ public void repositoryClass() throws CloneNotSupportedException {
         /* Insert Leonard */
         RestAssured.given()
                 .contentType(ContentType.JSON)
+                .queryParam("mappable", mappable)
                 .body(leonard)
                 .post("/nitrite/repositoryClass")
                 .then()
                 .statusCode(200)
                 .body("name", is("Leonard"));
-        RestAssured.get("/nitrite/repositoryClass")
+        RestAssured.given()
+                .queryParam("mappable", mappable)
+                .post("/nitrite/getRepositoryClass")
```

Comment:
This may fail because `getRepositoryClass` calls `consumerTemplate.receiveNoWait()` and there is some concurrency and i/o delays in game. A `consumerTemplate.receive*(...)` variant with timeout would perhaps be a better choice.

Reply:
@ppalaga  I think, that it is not necessary. Value is written into nitrite with previous post, and nitrite component waits until nitrite finishes: https://github.com/apache/camel/blob/master/components/camel-nitrite/src/main/java/org/apache/camel/component/nitrite/NitriteConsumer.java#L71
Then reading should work even without wait. WDYT? (to be sure, I also tried to "wait" in nitrite during insert, control is not released to camel component)

Reply:
You mean the consumer receives the data change event from Nitrite on the same thread where producer is writing the data? If so, yes, your setup may work well. But such an assumption sounds like an Nitrite's implementation detail that may change and it seems to be safer not to rely on that in our test.

Reply:
I'll refactor it as you suggested. It would make more sense, I can not be 100% sure, that this behavior will stay...

Reply:
fixed